### PR TITLE
Fixed metadata configuration for GCP

### PR DESCRIPTION
**Pull request by mangchiandjjoe**

### CODE REVIEW

File: start_esp/start_esp.py [hunk 1]
```diff
@@ -657,6 +664,15 @@ def make_argparser():
         Only works when --cors_preset is in use. Configures the CORS header
         Access-Control-Expose-Headers. Defaults to allow common response headers.
         ''')
+    parser.add_argument('--location',
+        help='''
+        One of acceptable zones, could not be any strings. Need to talk to
+        Chemist team to find out the acceptable zones.
+        If specified, send to ESP for location field and it will NOT be
+        overridden by Metadata server
+        ''')
```
Comment (qiwzhang) [inline start_esp/start_esp.py]: This is for us to talk to Chemist team.   Don't show in in the help.

File: start_esp/start_esp.py [hunk 2]
```diff
@@ -657,6 +664,15 @@ def make_argparser():
...
+    parser.add_argument('--non_gcp', action='store_true',
+        help='''Set the environment is non google cloud platform.''')
```
Comment (qiwzhang) [inline start_esp/start_esp.py]: By default, ESP tries to talk to GCP metadata server to get VM location in the first few requests.  setting this flag to true to skip this step.

File: start_esp/start_esp.py (hunk 1 above)
Comment (mangchiandjjoe) [reply start_esp/start_esp.py]: Removed location related changes

File: start_esp/start_esp.py (hunk 2 above)
Comment (mangchiandjjoe) [reply start_esp/start_esp.py]: Done

**Review by qiwzhang — APPROVED** [review]

**Pull request merged**